packages/dev/core/src/FlowGraph/Blocks/Data/Logic/flowGraphLogicBlocks.ts

```diff
@@ -0,0 +1,21 @@
+import { RichTypeBoolean } from "core/FlowGraph/flowGraphRichTypes";
+import { FlowGraphBinaryOperationBlock } from "../flowGraphBinaryOperationBlock";
+import { FlowGraphUnaryOperationBlock } from "../flowGraphUnaryOperationBlock";
+
+export class FlowGraphAndBlock extends FlowGraphBinaryOperationBlock<boolean, boolean, boolean> {
+    constructor() {
+        super(RichTypeBoolean, RichTypeBoolean, RichTypeBoolean, (left, right) => left && right);
+    }
+}
+
+export class FlowGraphOrBlock extends FlowGraphBinaryOperationBlock<boolean, boolean, boolean> {
+    constructor() {
+        super(RichTypeBoolean, RichTypeBoolean, RichTypeBoolean, (left, right) => left || right);
+    }
+}
+
+export class FlowGraphNotBlock extends FlowGraphUnaryOperationBlock<boolean, boolean> {

```

Comment:
it took me a few seconds to understand what you mean with this name :-)

I can't come up with a better one, I just think it might be confusing that there is a FlowGraphBlock and a FlowGraphNotBlock. Though I do understand it is a not-operation.

Reply:
Maybe FlowGraphLogicNotBlock? And rename and and or to FlowGraphAndLogic... etc?